onnxruntime/core/providers/nnapi/nnapi_builtin/builders/model_builder.cc

```diff
@@ -494,14 +490,13 @@ Status ModelBuilder::AddOperandFromPersistMemoryBuffer(
 
 Status ModelBuilder::AddOperations() {
   const auto& node_indices = graph_viewer_.GetNodesInTopologicalOrder();
-  std::unordered_set<const NodeUnit*> processed_node_units;
-  processed_node_units.reserve(node_unit_holder_.size());
-  for (size_t i = 0; i < node_indices.size(); i++) {
-    const auto* node(graph_viewer_.GetNode(node_indices[i]));
+  for (const auto node_idx : node_indices) {
+    LOGS_DEFAULT(VERBOSE) << "Adding node [" << node_idx << "]";
+    const auto* node(graph_viewer_.GetNode(node_idx));
     const NodeUnit& node_unit = GetNodeUnit(node);
 
-    // Since a NodeUnit may contain multiple nodes, avoid processing the same NodeUnit multiple times
-    if (Contains(processed_node_units, &node_unit))
+    // We only insert the NodeUnit once when we hit the target node
+    if (node != &node_unit.GetNode())
```

Comment:
> if (Contains(processed_node_units, &node_unit)) ...

just wondering why the logic seems to get changed here and removed the parts relating to caching the processed node units and such..?

Reply:
The red and blue node has same topological order, if red one (which is part of QDQConv group) was added before blue node, then it cannot find its input.
![image](https://user-images.githubusercontent.com/62914304/151632535-c329cbcd-aa24-4c54-a5f5-c320eb49500f.png)
Use target_node will guarantee the topological order of the node units

Reply:
I see. thx for the explanation!

Reply:
Added comments about this